[2024-06-20T08:21:35.053+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-20T08:21:35.156+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ml_pipeline.train_and_test_model manual__2024-06-20T08:20:46.488208+00:00 [queued]>
[2024-06-20T08:21:35.223+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ml_pipeline.train_and_test_model manual__2024-06-20T08:20:46.488208+00:00 [queued]>
[2024-06-20T08:21:35.226+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-06-20T08:21:35.318+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): train_and_test_model> on 2024-06-20 08:20:46.488208+00:00
[2024-06-20T08:21:35.347+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=640) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-20T08:21:35.354+0000] {standard_task_runner.py:63} INFO - Started process 642 to run task
[2024-06-20T08:21:35.353+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'ml_pipeline', 'train_and_test_model', 'manual__2024-06-20T08:20:46.488208+00:00', '--job-id', '79', '--raw', '--subdir', 'DAGS_FOLDER/ml_pipeline.py', '--cfg-path', '/tmp/tmp6woy7fpt']
[2024-06-20T08:21:35.361+0000] {standard_task_runner.py:91} INFO - Job 79: Subtask train_and_test_model
[2024-06-20T08:21:35.555+0000] {task_command.py:426} INFO - Running <TaskInstance: ml_pipeline.train_and_test_model manual__2024-06-20T08:20:46.488208+00:00 [running]> on host 12086e25b8f3
[2024-06-20T08:21:36.016+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='RCH4CKERS' AIRFLOW_CTX_DAG_ID='ml_pipeline' AIRFLOW_CTX_TASK_ID='train_and_test_model' AIRFLOW_CTX_EXECUTION_DATE='2024-06-20T08:20:46.488208+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-20T08:20:46.488208+00:00'
[2024-06-20T08:21:36.021+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-20T08:21:42.739+0000] {logging_mixin.py:188} INFO - Logistic Regression
[2024-06-20T08:21:42.748+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-20T08:21:42.750+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline.py", line 173, in train_and_test_model
    model = train_model(train_data)
            ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline.py", line 74, in train_model
    model = fit_model(model, x, y)
            ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline.py", line 98, in fit_model
    model.fit(X, y)
  File "/home/airflow/.local/lib/python3.12/site-packages/sklearn/base.py", line 1473, in wrapper
    return fit_method(estimator, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sklearn/linear_model/_logistic.py", line 1223, in fit
    X, y = self._validate_data(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sklearn/base.py", line 650, in _validate_data
    X, y = check_X_y(X, y, **check_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sklearn/utils/validation.py", line 1273, in check_X_y
    X = check_array(
        ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sklearn/utils/validation.py", line 1007, in check_array
    array = _asarray_with_order(array, order=order, dtype=dtype, xp=xp)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sklearn/utils/_array_api.py", line 746, in _asarray_with_order
    array = numpy.asarray(array, order=order, dtype=dtype)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/generic.py", line 2084, in __array__
    arr = np.asarray(values, dtype=dtype)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: could not convert string to float: '2014-12-13'
[2024-06-20T08:21:42.813+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=ml_pipeline, task_id=train_and_test_model, run_id=manual__2024-06-20T08:20:46.488208+00:00, execution_date=20240620T082046, start_date=20240620T082135, end_date=20240620T082142
[2024-06-20T08:21:42.874+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 79 for task train_and_test_model (could not convert string to float: '2014-12-13'; 642)
[2024-06-20T08:21:42.927+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-20T08:21:42.965+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-20T08:21:42.970+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
